src/mongoose_scram.erl

```diff
@@ -124,38 +138,79 @@ password_to_scram(Password) ->
     password_to_scram(Password, ?SCRAM_DEFAULT_ITERATION_COUNT).
 
 password_to_scram(#scram{} = Password, _) ->
+    %TODO: check if this is still correct
     Password;
 password_to_scram(Password, IterationCount) ->
     Salt = crypto:strong_rand_bytes(?SALT_LENGTH),
-    SaltedPassword = salted_password(Password, Salt, IterationCount),
-    StoredKey = stored_key(client_key(SaltedPassword)),
-    ServerKey = server_key(SaltedPassword),
-    #scram{storedkey = base64:encode(StoredKey),
-           serverkey = base64:encode(ServerKey),
-           salt = base64:encode(Salt),
-           iterationcount = IterationCount}.
+    ServerStoredKeys = [password_to_scram(Password, Salt, IterationCount, HashType)
+                            || {HashType, _Prefix} <- supported_sha_types()],
+    ResultList = lists:merge([{salt, base64:encode(Salt)},
+                              {iteration_count, IterationCount}], ServerStoredKeys),
+    maps:from_list(ResultList).
+
+password_to_scram(Password, Salt, IterationCount, HashType) ->
+    SaltedPassword = salted_password(HashType, Password, Salt, IterationCount),
+    StoredKey = stored_key(HashType, client_key(HashType, SaltedPassword)),
+    ServerKey = server_key(HashType, SaltedPassword),
+    {HashType, #{server_key => base64:encode(ServerKey),
+                 stored_key => base64:encode(StoredKey)}}.
 
-check_password(Password, Scram) ->
+check_password(Password, Scram) when is_record(Scram, scram)->
     IterationCount = Scram#scram.iterationcount,
     Salt = base64:decode(Scram#scram.salt),
     SaltedPassword = salted_password(Password, Salt, IterationCount),
     StoredKey = stored_key(client_key(SaltedPassword)),
-    (base64:decode(Scram#scram.storedkey) == StoredKey).
+    (base64:decode(Scram#scram.storedkey) == StoredKey);
+check_password(Password, ScramMap) when is_map(ScramMap) ->
+    #{salt := Salt, iteration_count := IterationCount} = ScramMap,
+    %TODO: change this check to verify for specific sha, not iterate over all supported
```

Comment:
Where would we need that? When the user authenticates with `PLAIN` auth method, we can probably just use the first supported and in the map scram. In the case of any `SCRAM-SHA-*` method, the `check_password` function is not used.